Dedicated filename input and default queue method (#964)
pymeasure/pymeasure: docs/tutorial/graphical.rst

      import pyqtgraph as pg
      pg.setConfigOption("useOpenGL", True)

The filename and directory input
################################

By default, a ManagedWindow instance contains fields for the filename and the directory to control where the results of an experiment are saved.

.. image:: pymeasure-fileinput.png
    :alt: The filename and directory input widget
    :width: 24%
.. image:: pymeasure-fileinput_disabled.png
    :alt: The filename and directory input widget, disabled to store the measurement to a temporary file
    :width: 24%
.. image:: pymeasure-fileinput_complete_filename.png
    :alt: The filename and directory input widget showing the auto-complete for the filename
    :width: 24%
.. image:: pymeasure-fileinput_complete_directory.png
    :alt: The filename and directory input widget showing the auto-complete for the directory
    :width: 24%

> With the filename input, additionally a toggle (named :guilabel:`Write file` controlling whether the measurement is save at all, or should be stored in a temporary file, is displayed.
> When the toggle is ticked, the measurement is stored to a temporary file, and the other input fields are disabled.

Reviewer: A more concise description
```suggestion
If the checkbox named :guilabel:`Write file` is enabled, the measurement is written to a file.
Otherwise, it is stored in a temporary file.
```

From the code below, I noticed an inconsistency:
You **un**check "write file" in order to write to a file?
For me it is clearer to tick "write file" to write it actually.
Alternatively, it could be labelled "temp file".

Author: Ah yeah, that is in inconsistency in the documentation; it is exactly as you say: you write to file when 'write file' is ticked

> When the measurement is to be stored in to a certain file, ensure that the :guilabel:`Write file` is disabled.

Reviewer: It doesn't seem right to me to check "write file" in order to **not** write to a file!

Author: Yes, you are right, this should reald "enabled". With your clarified sentence above, this one is unnecessary.

> The filename in the designated field can be entered with or without extension; if the entered extension is recognized (by default :code:`.csv` and :code:`.txt`) are recognized, that extension is used.

Reviewer: ```suggestion
The filename in the designated field can be entered with or without extension.
If the entered extension is recognized (by default :code:`.csv` and :code:`.txt` are recognized), that extension is used.
```

Parenthesis was misleading.

.. code-block:: python
   :emphasize-lines: 13, 14, 15, 16

    class MainWindow(ManagedWindow):

        def __init__(self):
            super().__init__(
                procedure_class=TestProcedure,
                inputs=['iterations', 'delay', 'seed'],
                displays=['iterations', 'delay', 'seed'],
                x_axis='Iteration',
                y_axis='Random Number',
            )
            self.setWindowTitle('GUI Example')

            self.filename = r'default_filename_delay{Delay Time:4f}s'   # Sets default filename
            self.directory = r'C:/Path/to/default/directory'            # Sets default directory
            self.store_measurement = False                              # Controls the 'write-file' togge
            self.file_input.extensions = ["csv", "txt", "data"]         # Sets recognized extensions, first entry is the default extension

The fields for the filename and the directory can be disabled using the boolean keyword arguments :code:`filename_input` and :code:`directory_input`, respectively, in the :class:`~pymeasure.display.windows.managed_window.ManagedWindow` init.

> Note that when a either of these values is set to :code:`False`, the default :meth:`~pymeasure.display.windows.managed_window.ManagedWindowBase.queue` method of the :class:`~pymeasure.display.windows.managed_window.ManagedWindow` class will no longer work, and a new, custom, method needs to be implemented; a basic implementation is shown in the documentation of the :meth:`~pymeasure.display.windows.managed_window.ManagedWindowBase.queue` method.

Reviewer: ```suggestion
Note that when either of these values is set to :code:`False`, the default :meth:`~pymeasure.display.windows.managed_window.ManagedWindowBase.queue` method of the :class:`~pymeasure.display.windows.managed_window.ManagedWindow` class will no longer work, and a new, custom, method needs to be implemented; a basic implementation is shown in the documentation of the :meth:`~pymeasure.display.windows.managed_window.ManagedWindowBase.queue` method.
```

This file can also be automatically loaded at the start of the program by adding the key-word argument :code:`sequence_file="filename.txt"` to the :code:`super().__init__` call, as was done in the example.

> The filename and directory input

Author: Note to self: modify this section.
